homeassistant/components/fan/__init__.py

```diff
@@ -116,6 +125,21 @@ async def async_set_speed(self, speed: str):
         """Set the speed of the fan."""
         if speed == SPEED_OFF:
             await self.async_turn_off()
+        elif self.supported_features & SUPPORT_SET_SPEED_PERCENTAGE:
+            await self.async_set_speed_percentage(self.speed_to_percentage(speed))
+        else:
+            await self.hass.async_add_job(self.set_speed, speed)
+
+    def set_speed_percentage(self, speed: str) -> None:
+        """Set the speed of the fan, as a percentage."""
+        raise NotImplementedError()
+
+    async def async_set_speed_percentage(self, speed: int):
+        """Set the speed of the fan, as a percentage."""
+        if speed == SPEED_OFF:
```

Comment:
This can't happen if it has to be an int.

Reply:
Thought I had changed that to 0, oops :)